cpp/src/command_classes/CentralScene.cpp

```diff
@@ -201,10 +234,52 @@ bool CentralScene::HandleMsg
 
 		if( Node* node = GetNodeUnsafe() )
 		{
-				char lbl[64];
-				for (int i = 1; i <= m_scenecount; i++) {
-					snprintf(lbl, 64, "Scene %d", i);
-					node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), _instance, i, lbl, "", true, false, 0, 0 );
+				for (int i = 1; i <= m_scenecount && i <= 31; i++) {
+					if ( version2orHigher )
+					{
+					    if ( identical )
+					    {
+					        // instance for scene number normaly isn't right, but in this case i think this is the best solution.
+					        // Because i higly doubt that someone will ever use the multichannel class together with the centralScene class.
+                            int keyAttributes = _data[3];
+                            if ( keyAttributes & CentralSceneMask_KeyPressed1time)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_SinglePress, "Scene", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed2times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_DoublePress, "Scene double press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_HeldDown)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_HeldDown, "Scene held down", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyReleased)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Released, "Scene released", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed3times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_TriplePress, "Scene triple press", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed4times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed4Times, "Scene pressed 4 times", "", true, false, 0, 0 );
+                            }
+                            if ( keyAttributes & CentralSceneMask_KeyPressed5times)
+                            {
+                                node->CreateValueInt(ValueID::ValueGenre_User, GetCommandClassId(), i, CentralScene_Pressed5Times, "Scene pressed 5 times", "", true, false, 0, 0 );
+                            }
+                        }
```

Comment:
Ok, sounds kinda good, i can have a look at this tomorrow.
One thing though: 
The class tells me exactly which keyAttributes (single / double press ... ) each scene supports. This information is lost with this approach? Like the Fibaro Button supports up to 5 presses, but the Nodon remotes only single and double presses (iirc).
For my part i'd like to know in my application what keypresses are supported.
Is there some other way to store this?